File: tessoku/B08.py
N = int(input())

grid = []
for i in range(1500):
  grid.append([0]*1500)
for i in range(N):
  x, y = map(int, input().split())
  grid[x-1][y-1] += 1

# print(grid)

sum_grid = []
for i in range(1501):
  sum_grid.append([0]*1501)

# 横方向累積和
for i in range(1, 1501):
  for j in range(1, 1501):
    sum_grid[i][j] = sum_grid[i][j-1] + grid[i-1][j-1]

# 縦方向累積和
for j in range(1, 1501):
  for i in range(1, 1501):
    sum_grid[i][j] = sum_grid[i-1][j] + sum_grid[i][j]

# print(sum_grid)

Q = int(input())
queries = []
for i in range(Q):
  a, b, c, d = map(int,input().split())
  queries.append([a, b, c, d])

for a, b, c, d in queries:
  ans = sum_grid[c][d] - sum_grid[a-1][d] -sum_grid[c][b-1] + sum_grid[a-1][b-1]
  print(ans)

File: tessoku_py/B08.cpp
#include <iostream>
using namespace std;

int N, X, Y, grid[1509][1509], sum_grid[1509][1509];
int Q, a, b, c, d;
int ans;

int main(){
  cin >> N;

  // 配列初期化
  for (int i=0; i<=1500; i++) {
    for (int j=0; j<=1500; j++) {
      grid[i][j] = 0;
      sum_grid[i][j] = 0;
    }
  }

  // 座標ごとに点の数を集計
  for (int i=1; i<=N; i++) {
    cin >> X >> Y;
    grid[X][Y] += 1;
  }

  // 二次元累積和作成
  for (int i=1; i<=1500; i++) {
    for (int j=1; j<=1500; j++) {
      sum_grid[i][j] = sum_grid[i][j-1] + grid[i][j];
    }
  }
  for (int j=1; j<=1500; j++) {
    for (int i=1; i<=1500; i++) {
      sum_grid[i][j] = sum_grid[i-1][j] + sum_grid[i][j];
    }
  }

  cin >> Q;
  for (int i=1; i<=Q; i++) {
    cin >> a >> b >> c >> d;
    ans = sum_grid[c][d]-sum_grid[c][b-1]-sum_grid[a-1][d]+sum_grid[a-1][b-1];
    cout << ans << endl;
  }

  return 0;
}